src/test/ui/proc-macro/inner-attrs.rs

```diff
@@ -0,0 +1,28 @@
+// check-pass
+// compile-flags: -Z span-debug --error-format human
+// aux-build:test-macros.rs
+
+#![feature(custom_inner_attributes)]
+#![feature(proc_macro_hygiene)]
+
+#[macro_use]
+extern crate test_macros;
+
+#[print_target_and_args(first)]
+#[print_target_and_args(second)]
+fn foo() {
+    #![print_target_and_args(third)]
+    #![print_target_and_args(fourth)]
+}
+
+fn bar() {
+    (#![print_target_and_args(fifth)] 1, 2);
```

Comment:
Could you add cases for other inner attributes in non-block expressions as well?
(Can be found by searching for `parse_inner_attributes`.)
```rust
[#![attr] 1, 2]

match 0 {
    #![attr]
    _ => {}
}

Struct { #![attr] field: 0 }
```